packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -280,16 +279,15 @@ function ensureListeningTo(
       rootContainerInstance.nodeType === COMMENT_NODE
         ? rootContainerInstance.parentNode
         : rootContainerInstance;
-    // Containers can only ever be element nodes. We do not
+    // Containers should only ever be element nodes. We do not
     // want to register events to document fragments or documents
     // with the modern plugin event system.
-    invariant(
+    if (
       rootContainerElement != null &&
-        rootContainerElement.nodeType === ELEMENT_NODE,
-      'ensureListeningTo(): received a container that was not an element node. ' +
-        'This is likely a bug in React.',
-    );
-    listenToEvent(registrationName, ((rootContainerElement: any): Element));
+      rootContainerElement.nodeType === ELEMENT_NODE
+    ) {
+      listenToEvent(registrationName, ((rootContainerElement: any): Element));
```

Comment:
We don't need an invariant here, we can use this more as a utility function. This let's us re-use this function and it's logic